src/main/java/http/request/RequestParser.java

```diff
@@ -1,15 +1,39 @@
 package http.request;
 
+import http.helper.Messages;
+import http.helper.Utils;
 
+import java.io.*;
+import java.nio.Buffer;
 
-import java.nio.charset.StandardCharsets;
-import java.util.*;
 
 public class RequestParser {
 
 
+    public static void requestReader(BufferedReader reader) throws IOException {
```

Comment:
It looks that what it's doing is reading the line and printing it out but calling it can be risky. When you call `reader.readLine()`, the line that is being read is consumed, we "lose" it (unless we store it or pass it somewhere as you do on line 29 of `Router`. I _think_ this is where the error (`Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2`) is coming from - the line 32 in `Router` is breaking (`requestReader(reader);` ) as there is nothing else to read. I _think_ you fixed it by adding try and catch, but it breaks and I assume it doesn't reach `Handler handler = fetchHandler(request);` ?

If you want to keep logging the request (keeping the functionality of `requestReader` in `RequestParser`), you can log the `parsedRequest` that you have in `Router`.